tests/Makefile/MakefileTest.php

```diff
@@ -0,0 +1,303 @@
+<?php
+/**
+ * This code is licensed under the BSD 3-Clause License.
+ *
+ * Copyright (c) 2017-2019, Maks Rafalko
+ * All rights reserved.
+ *
+ * Redistribution and use in source and binary forms, with or without
+ * modification, are permitted provided that the following conditions are met:
+ *
+ * * Redistributions of source code must retain the above copyright notice, this
+ *   list of conditions and the following disclaimer.
+ *
+ * * Redistributions in binary form must reproduce the above copyright notice,
+ *   this list of conditions and the following disclaimer in the documentation
+ *   and/or other materials provided with the distribution.
+ *
+ * * Neither the name of the copyright holder nor the names of its
+ *   contributors may be used to endorse or promote products derived from
+ *   this software without specific prior written permission.
+ *
+ * THIS SOFTWARE IS PROVIDED BY THE COPYRIGHT HOLDERS AND CONTRIBUTORS "AS IS"
+ * AND ANY EXPRESS OR IMPLIED WARRANTIES, INCLUDING, BUT NOT LIMITED TO, THE
+ * IMPLIED WARRANTIES OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE
+ * DISCLAIMED. IN NO EVENT SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE
+ * FOR ANY DIRECT, INDIRECT, INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL
+ * DAMAGES (INCLUDING, BUT NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR
+ * SERVICES; LOSS OF USE, DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER
+ * CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY,
+ * OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE
+ * OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+declare(strict_types=1);
+
+namespace Infection\Tests\Makefile;
+
+use function array_filter;
+use function array_shift;
+use function chdir;
+use function current;
+use function file_get_contents;
+use function getcwd;
+use function implode;
+use PHPUnit\Framework\TestCase;
+use function shell_exec;
+use function sprintf;
+use function strpos;
+use function substr_count;
+
+/**
+ * @coversNothing
+ *
+ * @group infra
+ */
+final class MakefileTest extends TestCase
+{
+    private const MAKEFILE_PATH = __DIR__ . '/../../Makefile';
+
+    /**
+     * @var string
+     */
+    private static $originalCwd;
+
+    /**
+     * {@inheritdoc}
+     */
+    public static function setUpBeforeClass(): void
+    {
+        self::$originalCwd = getcwd();
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    protected function tearDown(): void
+    {
+        chdir(self::$originalCwd);
+    }
+
+    public function test_the_default_goal_is_the_help_command(): void
+    {
+        $output = shell_exec(sprintf('make -f %s', self::MAKEFILE_PATH));
+
+        $expectedOutput = <<<'EOF'
```

Comment:
I did not find an easy way to have one test with the ANSI characters like here and another one without... If someone has an idea I'm all ears. Otherwise we can just keep carry one it's not a big deal